grafana/grafana
# InfluxDB: InfluxQL: allow empty tag values in the query editor

in the influxdb influxql query editor, you cannot represent this situation:
`FROM ... SELECT .. WHERE tag1<>""`

you cannot set the tag-value to empty-string. this pull-request tries to fix this. we simply use the builting support for this that exist in `@grafana/ui/Select`. this way, when you open the select-box for tag-values, there is a little [x] button on the line that allows clearing the value.

i had to update the query-building code too, to handle the case where tag-value is empty-string.

unfortunately this means that the `onChange` handlers now can be called with `null` (this is how the field-clearing is represented), so more lines of code needed to be added.


how to test:
- create an influxql query in the query-editor, add a WHERE-part with a tag-key and tag-value, then verify that you are able to clear the tag-value


NOTE:
there are two ways the selectboxes can work. the list is either filtered in the browser, or on the server. this means, in the first case, when the selectbox opens, we request the full list of options from the server, and then as you write in the field, we filter the list of options in the browser. in the second option, when you write something in the field, we send that string to the server to get a filtered list of options. this clearable-field-thing only works currently for filtered-in-the-browser cases, the reason being that we only need it there 😁 (for the WHERE-section for tag-values).

fixes https://github.com/grafana/grafana/issues/5600
